packages/domain/lib/use_case/auth_use_case.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:domain/model/firebase_user_email.dart';
+import 'package:domain/repository/auth_repository.dart';
+import 'package:domain/repository/preferences_local_repository.dart';
+import 'package:domain/use_case/sample_use_case/use_case_in_out.dart';
+
+
+class LoginEmailAndPassUseCase
+    extends UseCaseInOut<UserEmailPass, Future<bool>> {
+  final AuthRepository authRepository;
+  final PreferencesLocalRepository preferences;
+
+  LoginEmailAndPassUseCase(
+    this.authRepository,
+    this.preferences,
+  );
+
+  @override
+  Future<bool> call(UserEmailPass user) async {
+    final List<UserEmailPass> users = await authRepository.fetchUsers();
+    final isAbleToLogin = users.any(
+      (element) =>
+          element.login == user.login && element.password == user.password,
+    );
+    if (isAbleToLogin) await preferences.saveLoggedUser(user);
+    return isAbleToLogin;
+  }
```

Comment:
searching the user by login and password should be done inside the query. so, in this case u can make authRepository.fetchUsers() return boolean

but it's great, that u did it not by usual loop but with any operator!